server/route.go

```diff
@@ -1367,7 +1365,7 @@ func (c *client) processRemoteSub(argo []byte, hasOrigin bool) (err error) {
 	// If the account name is empty (not a "per-account" route), the account
 	// is at the index prior to the subject.
 	if accountName == _EMPTY_ {
-		accountName = string(args[subjIdx-1])
+		accountName = bytesToString(args[subjIdx-1])
```

Comment:
I think we should revert that one too, since you did few lines above for:
```
	c.mu.Lock()
	accountName := string(c.route.accName)
	c.mu.Unlock()
```
Again, we own the buffer here since we made a copy, but since the account name may be used to store as key to map, etc.. I am not sure if pinning this whole buffer just for the account name is not going to backfire in some cases.